java/ql/src/semmle/code/java/frameworks/apache/Collections.qll

```diff
@@ -138,127 +188,420 @@ private class ApacheKeyValueModel extends SummaryModelCsv {
+          ";IterableUtils;true;emptyIfNull;;;Argument[0];ReturnValue;value",
+          ";IterableUtils;true;filteredIterable;;;Element of Argument[0];Element of ReturnValue;value",
+          ";IterableUtils;true;find;;;Element of Argument[0];ReturnValue;value",
+          ";IterableUtils;true;first;;;Element of Argument[0];ReturnValue;value",
+          ";IterableUtils;true;get;;;Element of Argument[0];ReturnValue;value",
+          ";IterableUtils;true;loopingIterable;;;Element of Argument[0];Element of ReturnValue;value",
+          ";IterableUtils;true;partition;;;Element of Argument[0];Element of Element of ReturnValue;value",
+          ";IterableUtils;true;reversedIterable;;;Element of Argument[0];Element of ReturnValue;value",
+          ";IterableUtils;true;skippingIterable;;;Element of Argument[0];Element of ReturnValue;value",
+          ";IterableUtils;true;toList;;;Element of Argument[0];Element of ReturnValue;value",
+          ";IterableUtils;true;toString;;;Argument[2];ReturnValue;taint",
+          ";IterableUtils;true;toString;;;Argument[3];ReturnValue;taint",
+          ";IterableUtils;true;toString;;;Argument[4];ReturnValue;taint",
+          ";IterableUtils;true;uniqueIterable;;;Element of Argument[0];Element of ReturnValue;value",
+          ";IterableUtils;true;unmodifiableIterable;;;Element of Argument[0];Element of ReturnValue;value",
+          ";IterableUtils;true;zippingIterable;;;Element of Argument[0];Element of ReturnValue;value",
+          ";IterableUtils;true;zippingIterable;(Iterable,Iterable[]);;Element of ArrayElement of Argument[1];Element of ReturnValue;value",
+          ";IterableUtils;true;zippingIterable;(Iterable,Iterable);;Element of Argument[1];Element of ReturnValue;value"
+        ]
+  }
+}
+
+/**
+ * Value-propagating models for the class `org.apache.commons.collections4.EnumerationUtils`.
+ */
+private class ApacheEnumerationUtilsModel extends SummaryModelCsv {
+  override predicate row(string row) {
+    row =
+      ["org.apache.commons.collections4", "org.apache.commons.collections"] +
+        [
+          ";EnumerationUtils;true;get;;;Element of Argument[0];ReturnValue;value",
+          ";EnumerationUtils;true;toList;(Enumeration);;Element of Argument[0];Element of ReturnValue;value",
+          ";EnumerationUtils;true;toList;(StringTokenizer);;Argument[0];Element of ReturnValue;taint"
+        ]
+  }
+}
+
+/**
+ * Value-propagating models for the class `org.apache.commons.collections4.MultiMapUtils`.
+ */
+private class ApacheMultiMapUtilsModel extends SummaryModelCsv {
+  override predicate row(string row) {
+    row =
+      ["org.apache.commons.collections4", "org.apache.commons.collections"] +
+        [
+          ";MultiMapUtils;true;emptyIfNull;;;Argument[0];ReturnValue;value",
+          ";MultiMapUtils;true;getCollection;;;MapValue of Argument[0];ReturnValue;value",
```

Comment:
I think these need an `Element of` on both sides, because it's creating a new Collection/Bag/List/Set to hold whatever values the MultiValuedMap holds for the key

Reply:
I don't think so. The source code is literally `map.get(key)` behind a null check.

Reply:
Ah ok, I thought it stored them internally as something different.